src/_tutorial/chapter_5.rs

```diff
@@ -199,9 +199,23 @@
 //! }
 //! ```
 //!
-//! If you look closely at [`repeat`], it isn't collecting directly into a [`Vec`] but
-//! anything that implements the [`Accumulate`] trait to gather the results. This lets us make more complex parsers than we did in
-//! [`chapter_2`] by accumulating the results into a `()` and [`take`][Parser::take]-ing the captured input:
+//! If you look closely at [`separated`] and [`repeat`], they aren't limited to collecting
+//! the result into a [`Vec`], but rather anything that implements the [`Accumulate`] trait.
+//! [`Accumulate`] is for instance also implemented for [`HashSet`], [`String`] and `()`.
+//!
+//! This lets us build more complex parsers than we did in
+//! [`chapter_2`] by accumulating the results into a `()` and [`take`][Parser::take]-ing
+//! the consumed input.
+//!
+//! `take` works by creating a [`checkpoint`][Stream::checkpoint] before running the
+//! inner parser. In our case the `parse_list` parser then advances to the end of the digit list,
+//! afterwards `take` reads the offset (i.e. how many chars were matched by `parse_list`),
+//! restores the checkpoint, and returns the whole digit list as a single string based on
+//! the previously read offset.
+//!
+//! Due to this `take` doesn't care about the result of `parse_list`, which is why `parse_list`
+//! simply accumulates into a `()`, which effectively discards the product of `parse_list`.
```

Comment:
The sentence starting with "In our case" feels a bit off.  Also, I don't think we need to talk about offsets (and that invites going even deeper into the implementation weeds).  In trying to come up with an alternative, it made me wonder if a list would work better.

```suggestion
//! `take` works by
//! 1. Creating a [`checkpoint`][Stream::checkpoint]
//! 2. Running the inner parser, in our case the `parse_list` parser, which will advance the input
//! 3. Creating a [`checkpoint`][Stream::checkpoint] 
//! 4. Returning the slice from the first checkpoint to the second in place of the inner parser's result
//! 
//! Since the result of `parse_list` gets thrown away, we 
//! accumulates into a `()` to not waste work creating an unused `Vec`.
```

Thoughts?

Reply:
Looks good to me, makes it a lot easier to read and keep track of the steps.

Reply:
I assume you decided to create a second checkpoint for the sake of simplicity?

The code doesn't work that way and resets to the checkpoint before advancing again by the offset of `parse_list` and returning the slice.

Reply:
I wanted to gloss over the reset.

We could probably drop the second checkpoint and say we return the slice from the checkpoint to the current location.

Reply:
Yeah, I think I like that better :)
Adjusted it accordingly